src/main/java/shef/data/LiveStream.java

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shef.data;
+
+import java.util.Date;
+ 
+/** A YouTube live stream.*/
+public class LiveStream {
+  String userId;
+  String recipeKey;
+  String link;
+  Date startTime;
+  Date endTime;
+  long duration;
+
+ /**
+  * @param userId Unique id of the associated user (poster).
+  * @param recipeKey Unique key of the associated recipe.
+  * @param link Link to the live stream.
+  */
+  public LiveStream(String userId, String recipeKey, String link) {
```

Comment:
How are start and end times set during a new LiveStream construction?

Reply:
They are not - they're only used in the java servlets when interacting with data store. The user never sees them, so they're unnecessary to pass to the object. They're only used to order the entries, so they're necessary as fields in live stream (since the java servlet constructs these entries to send to the response, which prints the info so that js can use it for display), but they're otherwise never assigned.